api/src/main/java/org/apache/iceberg/actions/RewriteDataFiles.java

```diff
@@ -77,6 +77,17 @@
    */
   String TARGET_FILE_SIZE_BYTES = "target-file-size-bytes";
 
+  /**
+   * If the compaction should commit rewritten data files using the sequence number at compaction start time instead
+   * of optimistically incrementing the latest sequence number.
+   * <p>
+   * This avoids commit conflicts with updates that add newer equality deletes at a higher sequence number.
+   * <p>
+   * Defaults to false.
+   */
+  String USE_STARTING_SEQUENCE_NUMBER = "use-starting-sequence-number";
+  boolean USE_STARTING_SEQUENCE_NUMBER_DEFAULT = false;
```

Comment:
Is there a reason why we wouldn't use this as the default?

Reply:
+1. And I have some concerns. In what situation do users need to use the new sequence number when commit the rewritten data file instead of using the starting sequence number? Maybe this configuration is not necessary?

Reply:
The reason I used false here is because otherwise it changes the existing behavior that has been released. Are we allowed to change this behavior?

Reply:
I think it is fine to change this behavior. It doesn't affect correctness and it will avoid conflicts. That's a win. I also can't imagine people relying on the behavior of sequence numbers at this level.

Reply:
sounds good, I will update